plog-kafka/src/main/java/com/airbnb/plog/kafka/KafkaHandler.java

```diff
@@ -13,29 +13,31 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
-import java.io.ByteArrayOutputStream;
-import java.util.concurrent.atomic.AtomicLong;
-
 import javax.crypto.Cipher;
 import javax.crypto.spec.SecretKeySpec;
+import java.io.ByteArrayOutputStream;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
 
 @RequiredArgsConstructor
 @Slf4j
 public final class KafkaHandler extends SimpleChannelInboundHandler<Message> implements Handler {
     private final String defaultTopic;
     private final boolean propagate;
-    private final Producer<byte[], byte[]> producer;
+    private final Producer<Integer, byte[]> producer;
```

Comment:
agree, but the default partitioner use `key.hashcode`, and the hashcode of an array is `Object.hashcode`. Java primitives/objects legacy...
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/producer/DefaultPartitioner.scala#L28

so instead of customizing the partitioner to use [`Arrays.hashcode`](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#hashCode(byte[])), I decided to use an int.